src/game/utils/batch/report.rs

```diff
@@ -0,0 +1,296 @@
+    ip_ev: f32,
+    ip_eqr: f32,
+    oop_equity: f32,
+    oop_ev: f32,
+    oop_eqr: f32,
+    actions: Vec<f32>,
+    action_evs: Vec<f32>,
+}
+
+impl Display for AggRow {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut all_stats: Vec<f32> = vec![
+            self.ip_equity,
+            self.ip_ev,
+            self.ip_eqr,
+            self.oop_equity,
+            self.oop_ev,
+            self.oop_eqr,
+        ];
+        all_stats.append(&mut self.actions.clone());
+        all_stats.append(&mut self.action_evs.clone());
+
+        write!(formatter, "{},", flop_to_string(&self.flop))?;
+        fmt_floats(&all_stats, formatter)
+    }
+}
+
+/// Tree structure for computing aggregate reports
+/// The strucutre mirrors the action tree of the pertinent game
+/// Example use
+/// TODO
+pub struct AggActionTree {
+    prev_actions: Vec<Action>,
+    avail_actions: Vec<Action>,
+    // NOTE: |child_trees| <= |avail_actions|
+    // Because no child is created for terminating nodes
+    // Use non_terminating_avail_actions to get corresponding list
+    // of actions taken to reach child trees
+    child_trees: Vec<AggActionTree>,
+    data: Vec<AggRow>,
+}
+
+impl AggActionTree {
+    pub fn init_root() -> AggActionTree {
+        Self::init(Vec::new(), Vec::new())
+    }
+
+    fn init(prev_actions: Vec<Action>, avail_actions: Vec<Action>) -> AggActionTree {
+        AggActionTree {
+            prev_actions,
+            avail_actions,
+            child_trees: Vec::new(),
+            data: Vec::new(),
+        }
+    }
+
+    // current_dir = dir that report should be written to
+    pub fn write(&self, current_dir: &str, report_file_name: &str) -> std::io::Result<()> {
+        // Write report
+        let mut f = File::create_new(format!("{}/{}", current_dir, report_file_name))?;
+        writeln!(f, "{}", report_header(&self.avail_actions))?;
+        for row in &self.data {
+            writeln!(f, "{}", row)?;
+        }
+
+        Ok(())
+    }
+
+    // TODO currently using `write!`, which will not overwrite file
+    // Should allow to overwrite file with some force option
+    pub fn write_self_and_children(
+        &self,
+        current_dir: &str,
+        report_file_name: &str,
+    ) -> std::io::Result<()> {
+        self.write(current_dir, report_file_name)?;
+
+        for (child, action) in self
+            .child_trees
+            .iter()
+            .zip(self.non_terminating_avail_actions().iter())
+        {
+            create_dir(format!(
+                "{}/{}",
+                current_dir,
+                folder_name_from_action(*action)
+            ))?;
+            child.write_self_and_children(
+                format!("{}/{}", current_dir, folder_name_from_action(*action)).as_str(),
+                report_file_name,
+            )?;
+        }
+
+        Ok(())
+    }
+
+    pub fn update(&mut self, game: &mut PostFlopGame, &flop: &[u8; 3]) {
```

Comment:
It looks like this is recursively computing `AggRow`s for every node in the tree? I'm not totally sure what this is doing.

I'd say maybe:
1. `update` is not the best name as it's doing more than updating
2. could use a docstring

Reply:
Remamed and documented for better readability.